pyhelm/repo.py

```diff
@@ -2,23 +2,82 @@
 import itertools
 import os
 from git import Repo
+from urlparse import urlparse
 import requests
 import shutil
 import tarfile
 import tempfile
 import yaml
+import boto3.s3
+from botocore.exceptions import ClientError
 
 
+def _get_from_http(repo_url, file_url):
+    """Downloads the Chart's repo index from HTTP(S)"""
+    
+    if repo_url not in file_url:
+        file_url = os.path.join(repo_url, file_url)
+
+    index = requests.get(file_url)
+    return index.content
+
+def _get_from_s3(repo_url, file_url):
+    """Download the index / Chart from S3 bucket"""
+
+    s3_client = boto3.client('s3')
+
+    # NOTE(ljakimczuk): this is done for two
+    # reasons. First, it allows to use this
+    # function for either getting index.yaml
+    # or Chart. Second, at least the Chartmuseum-
+    # generated index.yaml may have the relative
+    # URLs (guess due to its multi-tenancy), so
+    # turning them into absolute is needed.
+    if repo_url not in file_url:
+        file_url = os.path.join(repo_url, file_url)
+    
+    file_url_parsed = urlparse(file_url)
+
+    try:
+        file_object = s3_client.get_object(
+            Bucket=file_url_parsed.netloc,
+            Key=file_url_parsed.path.strip('/'),
+        )
+
+        return file_object['Body'].read()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchBucket':
+            raise RuntimeError('%s repository not found' % file_url_parsed.netloc)
+        elif e.response['Error']['Code'] == 'NoSuchKey':
+            raise RuntimeError('%s not found in the repository' % file_url_parsed.path.strip('/'))
+        else:
+            pass
```

Comment:
What would you expect the behaviour to be in this case? I'd re-raise the exception

Reply:
I would actually expect the code to stop, so I totally agree with re-raising the exception. To be honest I don't remember why I placed the `pass` there, probably it's due to being sloppy..